src/configs/starship_root.rs

```diff
@@ -12,8 +13,12 @@ pub struct StarshipRootConfig {
     pub scan_timeout: u64,
     pub command_timeout: u64,
     pub add_newline: bool,
+    pub palette: Option<String>,
+    pub palettes: HashMap<String, Palette>,
```

Comment:
Serializing `None` as TOML can cause issues, because it doesn't have a native type for that.
```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    pub palettes: HashMap<String, Palette>,
```